### DISCUSSION TITLE
Float Switch Integration

### DISCUSSION THREAD

**Discussion by kmontagne** [tagyoureit/nodejs-poolController#830] [General]
Does nodejs-poolController and/or REM support taking actions based on input from devices like float switches? I want to trigger an action (e.g. stop a pump or close an Intellivalve) when a float switch is triggered. 

Here is my scenario. I have an Intelliecenter installation and I am already running njsPC on Raspberry Pi. I will soon install a Glacier cooler. The cooler works by taking in water from the primary pump after the filter. Some of the return flow is routed to it via an Intellivalve that is programmed to open when the cooler is running. The cooler works via evaporation. The water is sprayed down via a rotating wand through baffles into a basin. There is a float switch in the basin that triggers the cooler's pump which sends water back to the pool via a dedicated return line.

The problem with the cooler's design is that if its output pump fails, the water will overflow the basin and pool water will continue to spill out until the main pump is shutoff or runs out of water. One solution to this problem would be to install a float switch near the top of the basin that either closes the Intellivalve that sends water to the cooler or switches off the main pump.

**Comment by rstrouse** [comment]
REM can be used to read that float switch.  How much flow is going down these baffles are we talking a 1/2 inch or 4 inch line?

**Reply by kmontagne** [reply]
The lines in and out of the cooler are  1-1/2". The recommneded starting point for the cooler's nput line is to open the control valve around 13% and adjust until the wands are rotating at the suggested speed. The model I have has a nominal flow of 30 GPM. 

I am old programmer and have spent some time browsing through the  njsPC code. I have not looked at or used REM. Is there some doc,  example, or discussion thread that describes how  they work together and how I could implement the float valve solution?
